lib/web/apiserver.go

```diff
@@ -1395,9 +1395,17 @@ func (h *Handler) getWebConfig(w http.ResponseWriter, r *http.Request, p httprou
 		}
 	}
 
+	clusterFeatures, err := h.cfg.ProxyClient.GetFeatures(r.Context())
+
+	if err != nil {
+		h.log.WithError(err).Warn("cannot retrieve cluster features, client may get a stale feature set")
```

Comment:
Could just fail here instead. Thoughts?

Reply:
Yea, I think that it should be safe to just return an error from this err flow like in case of `newSessionCache` err flow.